File: app/src/main/java/com/example/clubmanager/data/repo/InMemoryRepo.java
package com.example.clubmanager.data.repo;

import com.example.clubmanager.data.models.Ping;
import com.example.clubmanager.data.models.Staff;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class InMemoryRepo implements Repo {
    private final List<Ping> pings = new ArrayList<>();
    private final List<Staff> staff = new ArrayList<>(Arrays.asList(
            new Staff("s1", "Alicia", "Bartender"),
            new Staff("s2", "Myles", "Security"),
            new Staff("s3", "Maddox", "Manager"),
            new Staff("s4", "Aaron", "Floor")
    ));
    private final List<String> roles = new ArrayList<>(Arrays.asList(
            "Bartender", "Security", "Manager", "Floor"
    ));

    @Override public void addPing(Ping ping) { pings.add(ping); }
    @Override public List<Ping> getPings() { return new ArrayList<>(pings); }
    @Override public List<Staff> getStaff() { return new ArrayList<>(staff); }
    @Override public List<String> getRoles() { return new ArrayList<>(roles); }
}

File: app/src/main/java/com/example/clubmanager/data/repo/InMemoryRepo.kt
package com.example.clubmanager.data.repo

import com.example.clubmanager.data.models.Ping
import com.example.clubmanager.data.models.Staff

class InMemoryRepo : Repo {
    private val staffList = mutableListOf(
        Staff("s1","Alex","Security",  true),
        Staff("s2","Bea","Bartender", true),
        Staff("s3","Cam","Server",    false),
        Staff("s4","Dre","Manager",   true),
    )
    private val pingList = mutableListOf<Ping>()

    override val staff: List<Staff> get() = staffList
    override val pings: List<Ping>  get() = pingList

    override fun addPing(p: Ping) {
        pingList.add(0, p) // newest first
    }

    override fun acknowledge(pingId: String, staffId: String) {
        pingList.firstOrNull { it.id == pingId }?.let { p ->
            if (!p.ackBy.contains(staffId)) p.ackBy.add(staffId)
        }
    }
}
